tests/models/user.py

```diff
@@ -0,0 +1,195 @@
+
+
+def _test_user_purging(store, member, member2, purge):
+    assert store.units.first().target_f == ""
+    assert store.get_max_unit_revision() == 1
+
+    # member updates first unit, adding a suggestion
+    _create_submission_and_suggestion(store, member)
+    assert store.get_max_unit_revision() == 3
+    assert store.units.count() == 1
+    assert store.units.first().target_f == "Hello, world UPDATED"
+    assert store.units.first().submitted_by == member
+    assert store.units.first().get_suggestions().count() == 1
+
+    # (evil) member2 updates with:
+    #   - changes 1st unit
+    #   - adds another unit
+    #   - adds another suggestion
+    filename = "tests/data/po/tutorial/en/tutorial_update_evil.po"
+    _create_submission_and_suggestion(store,
+                                      member2,
+                                      filename=filename,
+                                      suggestion="EVIL SUGGESTION")
+    assert store.get_max_unit_revision() == 5
+    assert store.units.count() == 2
+    assert store.units.first().get_suggestions().count() == 2
+    assert store.units.first().target_f == "Hello, world EVIL"
+    assert store.units.first().submitted_by == member2
+    assert store.units[1].target_f == "Goodbye, world EVIL"
+    assert store.units[1].submitted_by == member2
+
+    # purge evil member2
+    purge(member2)
+
+    # back to original update - although revision count has increased
+    assert store.get_max_unit_revision() == 6
+    assert store.units.count() == 1
+    assert store.units.first().get_suggestions().count() == 1
+    assert (store.units.first().get_suggestions().first().target_f
+            == 'SUGGESTION')
+    assert store.units.first().target_f == "Hello, world UPDATED"
+    assert store.units.first().submitted_by == member
+
+
+@pytest.mark.django_db
+def test_purge_user(en_tutorial_po, member, member2):
+    """Test purging user using `purge_user` function"""
+    _test_user_purging(en_tutorial_po, member, member2,
+                       lambda m: accounts.utils.purge_user(m))
+
+
+@pytest.mark.django_db
+def test_delete_purge_user(en_tutorial_po, member, member2):
+    """Test purging user using `User.delete(purge=True)`"""
+    _test_user_purging(en_tutorial_po, member, member2,
+                       lambda m: m.delete(purge=True))
+
+
+@pytest.mark.django_db
+def test_verify_user(member_with_email):
+    """Test verifying user using `verify_user` function"""
+
+    # Member is not currently verified
+    with pytest.raises(EmailAddress.DoesNotExist):
+        EmailAddress.objects.get(user=member_with_email, verified=True)
+
+    # Verify user
+    accounts.utils.verify_user(member_with_email)
+
+    # Get the verified email object
+    EmailAddress.objects.get(user=member_with_email, email="member@this.test",
+                             primary=True, verified=True)
+
+
+@pytest.mark.django_db
+def test_verify_user_without_existing_email(member):
```

Comment:
I think it is preferable to create a new user fixture instead of reusing `member`.